org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/format/experimental/XMLFormatterJoinCommentLinesTest.java

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.services.format.experimental;
+
+import static java.lang.System.lineSeparator;
+
+import org.eclipse.lemminx.XMLAssert;
+import org.eclipse.lemminx.commons.BadLocationException;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.TextEdit;
+import org.junit.jupiter.api.Test;
+
+/**
+ * XML experimental formatter services tests with join comment lines setting.
+ *
+ */
+public class XMLFormatterJoinCommentLinesTest {
+	@Test
+	public void testJoinCommentLines() throws BadLocationException {
+		String content = "<!--" + lineSeparator() + //
+				" line 1" + lineSeparator() + //
+				" " + lineSeparator() + //
+				" " + lineSeparator() + //
+				"   line 2" + lineSeparator() + //
+				" -->";
+		String expected = "<!-- line 1 line 2 -->";
+		SharedSettings settings = new SharedSettings();
+		settings.getFormattingSettings().setJoinCommentLines(true);
+		assertFormat(content, expected, settings);
```

Comment:
Declare expected text edit with te(...).

Reply:
End your test with 

`assertFormat(expected, expected, settings);`

to check that it will not generate some textedit if teh XML is already formatted. Do that for the other tests.